src/utils/middleware/dueDate.ts

```diff
@@ -0,0 +1,36 @@
+import { NextFunction, Request, Response } from "express";
+import { UTCISODateFromString } from "italia-ts-commons/lib/dates";
+import { ResponseErrorNotFound } from "italia-ts-commons/lib/responses";
+import { log } from "../logger";
+
+/**
+ * Due Date Middleware Factory
+ * To be applied to endpoint which are available only until a certain date.
+ * If the check fails, a 404 Not Found error is returned, meaning the resource is no longer available
+ *
+ * @param dueDate date until the resource is valid
+ */
+export function dueDateMiddleware(
```

Comment:
Bonus: Add tests for the function

Reply:
here: https://github.com/pagopa/io-backend/pull/747/commits/824d612e7c72eb7689fcd7e4548fae0a80dbe83c